File: Year 2/Semester 4/MPP/JavaFifthWeek/src/main/java/com/example/temasaptamana5/repository/IRepository.java
package com.example.temasaptamana5.repository;

import com.example.temasaptamana5.domain.Entity;

public interface IRepository<ID, E extends Entity<ID>>  {
    /**
     *
     * @param id of type ID (template)
     * @return found entity or null
     */
    E findById(ID id);
    Iterable<E> getAll();
    boolean add(E entity);
    boolean delete(ID id);
    boolean update(E entity);
}

File: Year 2/Semester 4/MPP/ProtobuffProjectCSharp/persistence/interfaces/IRepository.cs
using ProtobuffProject.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProtobuffProject.repository.interfaces
{
    public interface IRepository<ID, E> where E : Entity<ID>
    {
        List<E> GetAll();
        E FindById(ID id);
        bool Add(E entity);
        bool Delete(ID id);
        bool Update(E entity);
    }
}
